flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/functions/hive/HiveGenericUDF.java

```diff
@@ -40,10 +41,11 @@
 	private static final Logger LOG = LoggerFactory.getLogger(HiveGenericUDF.class);
 
 	private transient GenericUDF.DeferredObject[] deferredObjects;
+	private transient HiveShim hiveShim;
```

Comment:
Does this mean hiveShim is not needed after deserialization?

Reply:
shim is only needed at initialization at the planner. also, shim is not serializable.

Reply:
I created https://issues.apache.org/jira/browse/FLINK-13947 to investigate this.